src/Sylius/Component/Inventory/Operator/BackordersHandler.php

```diff
@@ -78,13 +84,16 @@ public function fillBackorders(StockableInterface $stockable)
             return;
         }
 
-        $units = $this->repository->findBy([
-            'stockable' => $stockable,
-            'inventoryState' => InventoryUnitInterface::STATE_BACKORDERED,
-        ], ['createdAt' => 'ASC']);
+        $units = $this->repository->findByStockableAndInventoryState(
+            $stockable,
+            InventoryUnitInterface::STATE_BACKORDERED,
+            $onHand
+        );
 
         foreach ($units as $unit) {
-            $unit->setInventoryState(InventoryUnitInterface::STATE_SOLD);
+            $this->factory->get($unit, InventoryUnitTransitions::GRAPH)
```

Comment:
It would be better to put each call in separated line. It would improve readability.